### PR TITLE
Add Risk of Rain 2

### LINKED ISSUE #886: Request: Risk of rain 2 dedicated server
---------------
Service: (Ex. minecraft/factorio/etc)
Risk of rain 2 dedicated server

Does this expand an already existing service: Y/N 
Yes this project allows you to host it anywhere instead of ONLY your local machine.(Official dedi: https://steamdb.info/app/1180760/)  

Link to game: (Ex. minecraft.net/factorio.com/etc) 
https://github.com/avivace/ror2-server

Links for server downloads: This needs to be an official link and not one that is hosted on some forum page or a personal github page.
(This is only what i could find it uses docker) https://hub.docker.com/r/avivace/ror2server 

Links for install steps/docs:
https://steamcommunity.com/sharedfiles/filedetails/?id=2077564253 (their github/docker page also has instructions)

### PR DESCRIPTION
**Pull request by acl1704**
### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

### New Server Submissions:

1. [ ] Does your submission pass tests (server is connectable)?
2. [x] Does your server use a custom docker image?
    * [x] Have you tried to use a generic image?
3. [x] Have you added the server to the main README.md?
4. [x] Have you added a unique README.md for the server you are adding?

Adds Risk of Rain 2 dedicated server.
Has not yet been tested - requires game ownership.

Closes #886

### CODE REVIEW

**Comment by parkervcp** [comment]
The server doesn't start.

```
:/home/container$ xvfb-run wine ./Risk of Rain 2.exe
wine: cannot find L"./Risk"
```

File: steamcmd_servers/ror2/egg-risk-of-rain2.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "images": [
+        "quay.io\/parkervcp\/pterodactyl-images:debian_wine-5"
+    ],
+    "file_denylist": [],
+    "startup": "xvfb-run wine .\/Risk of Rain 2.exe",
+    "config": {
+        "files": "{}",
+        "startup": "{\r\n    \"done\": \"Press Enter to chat.\"\r\n}",
+        "logs": "{}",
+        "stop": "stop"
```
Comment (parkervcp) [inline steamcmd_servers/ror2/egg-risk-of-rain2.json:21]: this is not a valid command to stop the server

File: steamcmd_servers/ror2/egg-risk-of-rain2.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "exported_at": "2021-04-17T12:39:54-07:00",
+    "name": "Risk of Rain 2",
+    "author": "alex.chang-lam@protonmail.com",
+    "description": "Risk of Rain 2 dedicated server.",
+    "features": null,
+    "images": [
+        "quay.io\/parkervcp\/pterodactyl-images:debian_wine-5"
+    ],
+    "file_denylist": [],
+    "startup": "xvfb-run wine .\/Risk of Rain 2.exe",
```
Comment (parkervcp) [inline steamcmd_servers/ror2/egg-risk-of-rain2.json:16]: fails to start the server.

```
:/home/container$ xvfb-run wine ./Risk of Rain 2.exe
wine: cannot find L"./Risk"
```

**Review by parkervcp — CHANGES_REQUESTED** [review]

File: steamcmd_servers/ror2/README.MD [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+- At least 1000M RAM
+- Minimum 4GB hard disk space
+
+## Server Ports
+
+The RoR2 server requires a single port for access.
+
+| Port  | default |
+|-------|---------|
+| Game  | 27015   |
```
Comment (parkervcp) [inline steamcmd_servers/ror2/README.MD:17]: should probably add a note about the steam query port.

File: steamcmd_servers/ror2/egg-risk-of-rain2.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    "config": {
+        "files": "{}",
```
Comment (parkervcp) [inline steamcmd_servers/ror2/egg-risk-of-rain2.json:18]: you can add a config for the server but make a new one called `server.cfg` that is a copy of `server_startup.cfg`. You can do this during the install easily.

You can use the `file` parser to replace lines in the config.

**Review by parkervcp — CHANGES_REQUESTED** [review]

File: steamcmd_servers/ror2/egg-risk-of-rain2.json (hunk 1 above)
Comment (acl1704) [reply steamcmd_servers/ror2/egg-risk-of-rain2.json:21]: I've tried ^C, quit, stop, none work. ^B and ^/ work, would that be better than requiring the kill button to stop the server?

File: steamcmd_servers/ror2/egg-risk-of-rain2.json (hunk 2 above)
Comment (acl1704) [reply steamcmd_servers/ror2/egg-risk-of-rain2.json]: Fixed. Edited start command didn't make it back to the nest

**Comment by acl1704** [comment]
The server crashes with a stack overflow when the custom configuration file is present, not sure why just yet. Appears to fully start up with the default.

**Comment by Software-Noob** [comment]
Your config parser does not parse it correctly, you forget quotes and semicolons. The password variable should have an input rule of `nullable` instead of required and the config parse can handle the double-quotes. The crashing is because custom config should only contain convars, not execute server, host, or delete users.

[Here is a fixed working egg](https://pastebin.com/raw/CzGaE0AM)

**Comment by Ignaciox** [comment]
Fixed egg works see  https://github.com/parkervcp/eggs/issues/886#issuecomment-850685868

**Comment by acl1704** [comment]
Closed in favor of #886

**Comment by Software-Noob** [comment]
> Closed in favor of #886

#886 is an open issue referencing the egg in my comment above. You can implement the changes and it will be ready for a merge.

**Comment by parkervcp** [comment]
@acl1704 just update you egg to use that json

**Pull request closed without merging**